lib/next_ls.ex

```diff
@@ -171,6 +173,66 @@ defmodule NextLS do
     {:reply, symbols, lsp}
   end
 
+  # TODO extract to a separate module
+  # TODO make this work for modules and structs
+  # TODO how to handle aliases?
+  # TODO can we do this in a single DB query?
+  def handle_request(%TextDocumentReferences{params: %{position: position, text_document: %{uri: uri}}} = request, lsp) do
+    import NextLS.DB.Query
+
+    file = URI.parse(uri).path
+    line = position.line + 1
+    col = position.character + 1
+
+    result =
+      dispatch(lsp.assigns.registry, :databases, fn databases ->
+        for {database_pid, _} <- databases do
+          query = ~Q"""
+          SELECT identifier, arity, type, module
+          FROM "references" as refs
+          WHERE refs.file = ?
+            AND refs.start_line <= ? AND refs.end_line >= ?
+            AND refs.start_column <= ? AND refs.end_column >= ?
+          ORDER BY refs.id ASC
+          LIMIT 1;
+          """
```

Comment:
one thing to consider is this should work if you trigger "find references" on a function/module definition _or_ a reference.

Might need to add end_line,end_col to the symbols table, so we can figure out if we're "inside" one.

Reply:
Yeah that makes sense. This is what I meant as well when talking about aliases earlier.

I think that we may be able to use `Code.Fragment.surround_context/3` to know what we are hovering when trying to find references: a function call, a module reference, a function definition, a module definition, etc. And then use this information to query the `references` table accordingly.

Reply:
Looks like we have enough information in the symbols and references table. I made an improvement that now works with function/module definitions and references.